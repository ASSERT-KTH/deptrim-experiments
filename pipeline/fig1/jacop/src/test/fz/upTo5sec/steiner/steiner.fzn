var set of 1..7: X_INTRODUCED_0;
var set of 1..7: X_INTRODUCED_1;
var set of 1..7: X_INTRODUCED_2;
var set of 1..7: X_INTRODUCED_3;
var set of 1..7: X_INTRODUCED_4;
var set of 1..7: X_INTRODUCED_5;
var set of 1..7: X_INTRODUCED_6;
var set of 1..7: X_INTRODUCED_7 ::var_is_introduced :: is_defined_var;
var 0..1: X_INTRODUCED_8 ::var_is_introduced :: is_defined_var;
var set of 1..7: X_INTRODUCED_9 ::var_is_introduced :: is_defined_var;
var 0..1: X_INTRODUCED_10 ::var_is_introduced :: is_defined_var;
var set of 1..7: X_INTRODUCED_11 ::var_is_introduced :: is_defined_var;
var 0..1: X_INTRODUCED_12 ::var_is_introduced :: is_defined_var;
var set of 1..7: X_INTRODUCED_13 ::var_is_introduced :: is_defined_var;
var 0..1: X_INTRODUCED_14 ::var_is_introduced :: is_defined_var;
var set of 1..7: X_INTRODUCED_15 ::var_is_introduced :: is_defined_var;
var 0..1: X_INTRODUCED_16 ::var_is_introduced :: is_defined_var;
var set of 1..7: X_INTRODUCED_17 ::var_is_introduced :: is_defined_var;
var 0..1: X_INTRODUCED_18 ::var_is_introduced :: is_defined_var;
var set of 1..7: X_INTRODUCED_19 ::var_is_introduced :: is_defined_var;
var 0..1: X_INTRODUCED_20 ::var_is_introduced :: is_defined_var;
var set of 1..7: X_INTRODUCED_21 ::var_is_introduced :: is_defined_var;
var 0..1: X_INTRODUCED_22 ::var_is_introduced :: is_defined_var;
var set of 1..7: X_INTRODUCED_23 ::var_is_introduced :: is_defined_var;
var 0..1: X_INTRODUCED_24 ::var_is_introduced :: is_defined_var;
var set of 1..7: X_INTRODUCED_25 ::var_is_introduced :: is_defined_var;
var 0..1: X_INTRODUCED_26 ::var_is_introduced :: is_defined_var;
var set of 1..7: X_INTRODUCED_27 ::var_is_introduced :: is_defined_var;
var 0..1: X_INTRODUCED_28 ::var_is_introduced :: is_defined_var;
var set of 1..7: X_INTRODUCED_29 ::var_is_introduced :: is_defined_var;
var 0..1: X_INTRODUCED_30 ::var_is_introduced :: is_defined_var;
var set of 1..7: X_INTRODUCED_31 ::var_is_introduced :: is_defined_var;
var 0..1: X_INTRODUCED_32 ::var_is_introduced :: is_defined_var;
var set of 1..7: X_INTRODUCED_33 ::var_is_introduced :: is_defined_var;
var 0..1: X_INTRODUCED_34 ::var_is_introduced :: is_defined_var;
var set of 1..7: X_INTRODUCED_35 ::var_is_introduced :: is_defined_var;
var 0..1: X_INTRODUCED_36 ::var_is_introduced :: is_defined_var;
var set of 1..7: X_INTRODUCED_37 ::var_is_introduced :: is_defined_var;
var 0..1: X_INTRODUCED_38 ::var_is_introduced :: is_defined_var;
var set of 1..7: X_INTRODUCED_39 ::var_is_introduced :: is_defined_var;
var 0..1: X_INTRODUCED_40 ::var_is_introduced :: is_defined_var;
var set of 1..7: X_INTRODUCED_41 ::var_is_introduced :: is_defined_var;
var 0..1: X_INTRODUCED_42 ::var_is_introduced :: is_defined_var;
var set of 1..7: X_INTRODUCED_43 ::var_is_introduced :: is_defined_var;
var 0..1: X_INTRODUCED_44 ::var_is_introduced :: is_defined_var;
var set of 1..7: X_INTRODUCED_45 ::var_is_introduced :: is_defined_var;
var 0..1: X_INTRODUCED_46 ::var_is_introduced :: is_defined_var;
var set of 1..7: X_INTRODUCED_47 ::var_is_introduced :: is_defined_var;
var 0..1: X_INTRODUCED_48 ::var_is_introduced :: is_defined_var;
var 0..7: X_INTRODUCED_50 ::var_is_introduced :: is_defined_var;
var 0..7: X_INTRODUCED_51 ::var_is_introduced :: is_defined_var;
var 0..7: X_INTRODUCED_52 ::var_is_introduced :: is_defined_var;
var 0..7: X_INTRODUCED_53 ::var_is_introduced :: is_defined_var;
var 0..7: X_INTRODUCED_54 ::var_is_introduced :: is_defined_var;
var 0..7: X_INTRODUCED_55 ::var_is_introduced :: is_defined_var;
var 0..7: X_INTRODUCED_56 ::var_is_introduced :: is_defined_var;
array [1..7] of var set of int: Sets:: output_array([1..7]) = [X_INTRODUCED_0,X_INTRODUCED_1,X_INTRODUCED_2,X_INTRODUCED_3,X_INTRODUCED_4,X_INTRODUCED_5,X_INTRODUCED_6];
constraint set_le(X_INTRODUCED_1,X_INTRODUCED_0);
constraint set_le(X_INTRODUCED_2,X_INTRODUCED_1);
constraint set_le(X_INTRODUCED_3,X_INTRODUCED_2);
constraint set_le(X_INTRODUCED_4,X_INTRODUCED_3);
constraint set_le(X_INTRODUCED_5,X_INTRODUCED_4);
constraint set_le(X_INTRODUCED_6,X_INTRODUCED_5);
constraint int_le(X_INTRODUCED_8,1);
constraint int_le(X_INTRODUCED_10,1);
constraint int_le(X_INTRODUCED_12,1);
constraint int_le(X_INTRODUCED_14,1);
constraint int_le(X_INTRODUCED_16,1);
constraint int_le(X_INTRODUCED_18,1);
constraint int_le(X_INTRODUCED_20,1);
constraint int_le(X_INTRODUCED_22,1);
constraint int_le(X_INTRODUCED_24,1);
constraint int_le(X_INTRODUCED_26,1);
constraint int_le(X_INTRODUCED_28,1);
constraint int_le(X_INTRODUCED_30,1);
constraint int_le(X_INTRODUCED_32,1);
constraint int_le(X_INTRODUCED_34,1);
constraint int_le(X_INTRODUCED_36,1);
constraint int_le(X_INTRODUCED_38,1);
constraint int_le(X_INTRODUCED_40,1);
constraint int_le(X_INTRODUCED_42,1);
constraint int_le(X_INTRODUCED_44,1);
constraint int_le(X_INTRODUCED_46,1);
constraint int_le(X_INTRODUCED_48,1);
constraint int_eq(X_INTRODUCED_50,3);
constraint int_eq(X_INTRODUCED_51,3);
constraint int_eq(X_INTRODUCED_52,3);
constraint int_eq(X_INTRODUCED_53,3);
constraint int_eq(X_INTRODUCED_54,3);
constraint int_eq(X_INTRODUCED_55,3);
constraint int_eq(X_INTRODUCED_56,3);
constraint set_intersect(X_INTRODUCED_0,X_INTRODUCED_1,X_INTRODUCED_7):: defines_var(X_INTRODUCED_7);
constraint set_card(X_INTRODUCED_7,X_INTRODUCED_8):: defines_var(X_INTRODUCED_8);
constraint set_intersect(X_INTRODUCED_0,X_INTRODUCED_2,X_INTRODUCED_9):: defines_var(X_INTRODUCED_9);
constraint set_card(X_INTRODUCED_9,X_INTRODUCED_10):: defines_var(X_INTRODUCED_10);
constraint set_intersect(X_INTRODUCED_0,X_INTRODUCED_3,X_INTRODUCED_11):: defines_var(X_INTRODUCED_11);
constraint set_card(X_INTRODUCED_11,X_INTRODUCED_12):: defines_var(X_INTRODUCED_12);
constraint set_intersect(X_INTRODUCED_0,X_INTRODUCED_4,X_INTRODUCED_13):: defines_var(X_INTRODUCED_13);
constraint set_card(X_INTRODUCED_13,X_INTRODUCED_14):: defines_var(X_INTRODUCED_14);
constraint set_intersect(X_INTRODUCED_0,X_INTRODUCED_5,X_INTRODUCED_15):: defines_var(X_INTRODUCED_15);
constraint set_card(X_INTRODUCED_15,X_INTRODUCED_16):: defines_var(X_INTRODUCED_16);
constraint set_intersect(X_INTRODUCED_0,X_INTRODUCED_6,X_INTRODUCED_17):: defines_var(X_INTRODUCED_17);
constraint set_card(X_INTRODUCED_17,X_INTRODUCED_18):: defines_var(X_INTRODUCED_18);
constraint set_intersect(X_INTRODUCED_1,X_INTRODUCED_2,X_INTRODUCED_19):: defines_var(X_INTRODUCED_19);
constraint set_card(X_INTRODUCED_19,X_INTRODUCED_20):: defines_var(X_INTRODUCED_20);
constraint set_intersect(X_INTRODUCED_1,X_INTRODUCED_3,X_INTRODUCED_21):: defines_var(X_INTRODUCED_21);
constraint set_card(X_INTRODUCED_21,X_INTRODUCED_22):: defines_var(X_INTRODUCED_22);
constraint set_intersect(X_INTRODUCED_1,X_INTRODUCED_4,X_INTRODUCED_23):: defines_var(X_INTRODUCED_23);
constraint set_card(X_INTRODUCED_23,X_INTRODUCED_24):: defines_var(X_INTRODUCED_24);
constraint set_intersect(X_INTRODUCED_1,X_INTRODUCED_5,X_INTRODUCED_25):: defines_var(X_INTRODUCED_25);
constraint set_card(X_INTRODUCED_25,X_INTRODUCED_26):: defines_var(X_INTRODUCED_26);
constraint set_intersect(X_INTRODUCED_1,X_INTRODUCED_6,X_INTRODUCED_27):: defines_var(X_INTRODUCED_27);
constraint set_card(X_INTRODUCED_27,X_INTRODUCED_28):: defines_var(X_INTRODUCED_28);
constraint set_intersect(X_INTRODUCED_2,X_INTRODUCED_3,X_INTRODUCED_29):: defines_var(X_INTRODUCED_29);
constraint set_card(X_INTRODUCED_29,X_INTRODUCED_30):: defines_var(X_INTRODUCED_30);
constraint set_intersect(X_INTRODUCED_2,X_INTRODUCED_4,X_INTRODUCED_31):: defines_var(X_INTRODUCED_31);
constraint set_card(X_INTRODUCED_31,X_INTRODUCED_32):: defines_var(X_INTRODUCED_32);
constraint set_intersect(X_INTRODUCED_2,X_INTRODUCED_5,X_INTRODUCED_33):: defines_var(X_INTRODUCED_33);
constraint set_card(X_INTRODUCED_33,X_INTRODUCED_34):: defines_var(X_INTRODUCED_34);
constraint set_intersect(X_INTRODUCED_2,X_INTRODUCED_6,X_INTRODUCED_35):: defines_var(X_INTRODUCED_35);
constraint set_card(X_INTRODUCED_35,X_INTRODUCED_36):: defines_var(X_INTRODUCED_36);
constraint set_intersect(X_INTRODUCED_3,X_INTRODUCED_4,X_INTRODUCED_37):: defines_var(X_INTRODUCED_37);
constraint set_card(X_INTRODUCED_37,X_INTRODUCED_38):: defines_var(X_INTRODUCED_38);
constraint set_intersect(X_INTRODUCED_3,X_INTRODUCED_5,X_INTRODUCED_39):: defines_var(X_INTRODUCED_39);
constraint set_card(X_INTRODUCED_39,X_INTRODUCED_40):: defines_var(X_INTRODUCED_40);
constraint set_intersect(X_INTRODUCED_3,X_INTRODUCED_6,X_INTRODUCED_41):: defines_var(X_INTRODUCED_41);
constraint set_card(X_INTRODUCED_41,X_INTRODUCED_42):: defines_var(X_INTRODUCED_42);
constraint set_intersect(X_INTRODUCED_4,X_INTRODUCED_5,X_INTRODUCED_43):: defines_var(X_INTRODUCED_43);
constraint set_card(X_INTRODUCED_43,X_INTRODUCED_44):: defines_var(X_INTRODUCED_44);
constraint set_intersect(X_INTRODUCED_4,X_INTRODUCED_6,X_INTRODUCED_45):: defines_var(X_INTRODUCED_45);
constraint set_card(X_INTRODUCED_45,X_INTRODUCED_46):: defines_var(X_INTRODUCED_46);
constraint set_intersect(X_INTRODUCED_5,X_INTRODUCED_6,X_INTRODUCED_47):: defines_var(X_INTRODUCED_47);
constraint set_card(X_INTRODUCED_47,X_INTRODUCED_48):: defines_var(X_INTRODUCED_48);
constraint set_card(X_INTRODUCED_0,X_INTRODUCED_50):: defines_var(X_INTRODUCED_50);
constraint set_card(X_INTRODUCED_1,X_INTRODUCED_51):: defines_var(X_INTRODUCED_51);
constraint set_card(X_INTRODUCED_2,X_INTRODUCED_52):: defines_var(X_INTRODUCED_52);
constraint set_card(X_INTRODUCED_3,X_INTRODUCED_53):: defines_var(X_INTRODUCED_53);
constraint set_card(X_INTRODUCED_4,X_INTRODUCED_54):: defines_var(X_INTRODUCED_54);
constraint set_card(X_INTRODUCED_5,X_INTRODUCED_55):: defines_var(X_INTRODUCED_55);
constraint set_card(X_INTRODUCED_6,X_INTRODUCED_56):: defines_var(X_INTRODUCED_56);
solve :: set_search(Sets,first_fail,indomain_min,complete) satisfy;
