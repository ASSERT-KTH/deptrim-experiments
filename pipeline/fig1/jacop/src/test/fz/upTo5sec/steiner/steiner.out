Sets = array1d(1..7, [{3, 5, 6}, {3, 4, 7}, {2, 5, 7}, {2, 4, 6}, {1, 6, 7}, {1, 4, 5}, 1..3]);
----------
